sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -266,25 +305,25 @@ def system_properties(self):
 
         if self._sys_properties is None:
             self._sys_properties = {}
-            if self.message.properties:
+            if self._raw_amqp_message.properties:
                 for key, prop_name in _SYS_PROP_KEYS_TO_MSG_PROPERTIES:
-                    value = getattr(self.message.properties, prop_name, None)
+                    value = getattr(self._raw_amqp_message.properties, prop_name, None)
                     if value:
                         self._sys_properties[key] = value
-            self._sys_properties.update(self.message.annotations)
+            self._sys_properties.update(self._raw_amqp_message.annotations)
         return self._sys_properties
 
     @property
     def body(self):
-        # type: () -> Union[bytes, Iterable[bytes]]
-        """The content of the event.
-
-        :rtype: bytes or Generator[bytes]
+        # type: () -> Any
+        """The body of the Message. The format may vary depending on the body type:
+        For ~azure.eventhub.AmqpMessageBodyType.DATA, the body could be bytes or Iterable[bytes]
+        For ~azure.eventhub.AmqpMessageBodyType.SEQUENCE, the body could be List or Iterable[List]
+        For ~azure.eventhub.AmqpMessageBodyType.VALUE, the body could be any type.
+        :rtype: Any
         """
-        try:
-            return self.message.get_data()
-        except TypeError:
-            raise ValueError("Event content empty.")
+
+        return self._raw_amqp_message.body
```

Comment:
this seems like a breaking change -- previously error is raised but now the data would be passed to user directly, so their handling logic might fail.

we'd better check when `get_data` would raise TypeError (if in case of sequence/value data, or in the case `get_data` returns None etc.

then decide whether we could simply `return self._raw_amqp_message.body` or we want to keep the error behavior

Reply:
looks like return type of `self._raw_amqp_message.body` is `Any`, so re-adding the exception